app/src/main/java/com/example/tasky/agenda/presentation/agenda_detail/AgendaDetailViewModel.kt

```diff
@@ -42,5 +47,56 @@ class AgendaDetailViewModel(
     private val eventChannel = Channel<AgendaEvent>()
     val events = eventChannel.receiveAsFlow()
 
-    fun onAction(action: AgendaAction) {}
+    fun onAction(action: AgendaDetailAction) {
+        when(action) {
+           is AgendaDetailAction.OnAgendaReminderIntervalSelect -> onAgendaReminderIntervalSelect(
+               reminder = action.reminder
+           )
+           is AgendaDetailAction.OnTimeFromPick -> {
+               val updatedTimestamp = updateUtcTime(currentTimestamp = _state.value.timestamp,
+                   hour = action.hour, minute = action.minute)
+               _state.update {
+                   it.copy(
+                       timestamp = updatedTimestamp
+                   )
+               }
+           }
+            is AgendaDetailAction.OnDateFromPick -> {
+                val updatedTimestamp = updateUtcDate(currentTimestamp = _state.value.timestamp,
+                    dateMillis = action.dateMillis)
+                _state.update {
+                    it.copy(
+                        timestamp = updatedTimestamp
+                    )
+                }
+            }
+        }
+    }
+
+    private fun onAgendaReminderIntervalSelect(reminder: AgendaReminderInterval) {
+        _state.update {
+            it.copy(selectedAgendaReminderInterval = reminder)
+        }
+    }
+
+    private fun updateUtcTime(currentTimestamp: ZonedDateTime, hour: Int, minute: Int): ZonedDateTime {
+        val localTime = LocalTime.of(hour, minute)
+        val localDate = currentTimestamp.toLocalDate()
+        val localDateTime = ZonedDateTime.of(localDate, localTime, ZoneId.systemDefault())
+        val utcDateTime = localDateTime.withZoneSameInstant(ZoneId.of("UTC"))
+        return utcDateTime
+    }
```

Comment:
This function looks more complex than it needs to be with all the intermediary dates and times being created. This should be equivalent if I'm not wrong:
```kotlin
return currentTimestamp
    .withHour(hour)
    .withMinute(minute)
    .withZoneSameInstant(ZoneId.of("UTC"))
```
This assumes that `currentTimestamp` is in the local user's timezone.